building/src/ts/components/building-toggle.tsx

```diff
@@ -0,0 +1,51 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import * as React from 'react';
+import {client, events} from 'camelot-unchained';
+
+export interface BuildingToggleProps {
+};
+
+export interface BuildingToggleState {
+  buildingIsActive: boolean;
+};
+
+declare const cuAPI: any;
+
+export class BuildingToggle extends React.Component<BuildingToggleProps, BuildingToggleState> {
+  constructor(props: BuildingToggleProps) {
+    super(props);
+    this.state = {
+      buildingIsActive: false
+    }
+  }
+
+  componentWillMount() {
+    this.setState({
+      buildingIsActive: false
+    });
+  }
+
+  toggleBuilding = (): void => {
+    this.setState({ buildingIsActive: !this.state.buildingIsActive } as any);
+    cuAPI.ToggleBuildingMode();
```

Comment:
you could use `client.ToggleBuildingMode()` here instead of direct access to `cuAPI`